src/category_theory/essential_image.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+import category_theory.natural_isomorphism
+
+/-!
+# Essential image of a functor
+
+The essential image of a functor consists of the objects in the target category which are isomorphic
+to an object in the image of the object function.
+This, for instance, allows us to talk about objects belonging to a subcategory expressed as a
```

Comment:
So, if I understand correctly, you are intentionally not putting a category structure on `ess_image` (as full subcategory of the target). Is that right? If so, please add it explicitly to this module docstring.

Reply:
It's more that `ess_image` is intended to be used as a predicate rather than as a subtype (or subcategory), but there's no particular reason not to have the subcategory - I'll add that.

Reply:
Done, with a little import shuffling